BUG: `signal.windows.kaiser_bessel_derived` uses `array` and is un-tested for `M<1`

### Describe your issue.

`signal.windows..kaiser_bessel_derived` uses `array` constructor for the case `M<1`, which is not a part of the array API standard -

https://github.com/scipy/scipy/blob/27157ac1db4fc23bef76df50dfd8a4393453153c/scipy/signal/windows/_windows.py#L1382

The reason that the array API tests running with `array_api_strict` don't error out is because the `M<1` case is never tested -

https://github.com/scipy/scipy/blob/27157ac1db4fc23bef76df50dfd8a4393453153c/scipy/signal/tests/test_windows.py#L504-L529

I'll be happy to pick this up!

### Reproducing Code Example

```python
# add the following to `TestKaiserBesselDerived.test_basic` and run the test
w = windows.kaiser_bessel_derived(0.5, beta=4.0, xp=xp)
xp_assert_equal(w, xp.asarray([]))
```

### Error message

```shell
________________________________________________________ TestKaiserBesselDerived.test_basic[array_api_strict] _________________________________________________________

self = <scipy.signal.tests.test_windows.TestKaiserBesselDerived object at 0x121183750>
xp = <module 'array_api_strict' from '/Users/saransh/Code/scipy/.env/lib/python3.13/site-packages/array_api_strict/__init__.py'>

    def test_basic(self, xp):
>       w = windows.kaiser_bessel_derived(0.5, beta=4.0, xp=xp)

self       = <scipy.signal.tests.test_windows.TestKaiserBesselDerived object at 0x121183750>
xp         = <module 'array_api_strict' from '/Users/saransh/Code/scipy/.env/lib/python3.13/site-packages/array_api_strict/__init__.py'>

scipy/signal/tests/test_windows.py:505: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
scipy/signal/windows/_windows.py:1382: in kaiser_bessel_derived
    return xp.array([])
        M          = 0.5
        beta       = 4.0
        device     = None
        sym        = True
        xp         = <module 'array_api_strict' from '/Users/saransh/Code/scipy/.env/lib/python3.13/site-packages/array_api_strict/__init__.py'>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'array'

    def __getattr__(name):
        if name in ['linalg', 'fft']:
            if name in get_array_api_strict_flags()['enabled_extensions']:
                if name == 'linalg':
                    from . import _linalg
                    return _linalg
                elif name == 'fft':
                    from . import _fft
                    return _fft
            else:
                raise AttributeError(f"The {name!r} extension has been disabled for array_api_strict")
>       raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
E       AttributeError: module 'array_api_strict' has no attribute 'array'. Did you mean: 'asarray'?

name       = 'array'

.env/lib/python3.13/site-packages/array_api_strict/__init__.py:354: AttributeError
======================================================================= short test summary info =======================================================================
FAILED scipy/signal/tests/test_windows.py::TestKaiserBesselDerived::test_basic[array_api_strict] - AttributeError: module 'array_api_strict' has no attribute 'array'. Did you mean: 'asarray'?
```

### SciPy/NumPy/Python version and system information

```shell
1.16.0.dev0+git20250404.753e10d 2.2.4 sys.version_info(major=3, minor=13, micro=2, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /opt/homebrew/Cellar/openblas/0.3.29/include
    lib directory: /opt/homebrew/Cellar/openblas/0.3.29/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=1 VORTEX MAX_THREADS=56
    pc file directory: /opt/homebrew/opt/openblas/lib/pkgconfig
    version: 0.3.29
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /opt/homebrew/Cellar/openblas/0.3.29/include
    lib directory: /opt/homebrew/Cellar/openblas/0.3.29/lib
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=1 VORTEX MAX_THREADS=56
    pc file directory: /opt/homebrew/opt/openblas/lib/pkgconfig
    version: 0.3.29
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.13.6
Compilers:
  c:
    commands: /opt/homebrew/opt/llvm/bin/clang
    linker: ld64
    name: clang
    version: 20.1.1
  c++:
    commands: /opt/homebrew/opt/llvm/bin/clang++
    linker: ld64
    name: clang
    version: 20.1.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.12
  fortran:
    commands: gfortran
    linker: ld64
    name: gcc
    version: 14.2.0
  pythran:
    include directory: ../.env/lib/python3.13/site-packages/pythran
    version: 0.17.0
Machine Information:
  build:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
  cross-compiled: false
  host:
    cpu: aarch64
    endian: little
    family: aarch64
    system: darwin
Python Information:
  path: /Users/saransh/Code/scipy/.env/bin/python3
  version: '3.13'
```

--- a/scipy/signal/tests/test_windows.py
+++ b/scipy/signal/tests/test_windows.py
@@ -502,7 +502,7 @@ class TestKaiser:
 class TestKaiserBesselDerived:
 
     def test_basic(self, xp):
-        # cover case `M < 1`
+        # cover case `M < 1`
         w = windows.kaiser_bessel_derived(0.5, beta=4.0, xp=xp)
         xp_assert_equal(w, xp.asarray([]))
 

--- a/scipy/signal/tests/test_windows.py
+++ b/scipy/signal/tests/test_windows.py
@@ -502,6 +502,9 @@ class TestKaiser:
 class TestKaiserBesselDerived:
 
     def test_basic(self, xp):
+        w = windows.kaiser_bessel_derived(0.5, beta=4.0, xp=xp)
+        xp_assert_equal(w, xp.asarray([]))
+
         M = 100
         w = windows.kaiser_bessel_derived(M, beta=4.0, xp=xp)
         w2 = windows.get_window(('kaiser bessel derived', 4.0),

--- a/scipy/signal/tests/test_windows.py
+++ b/scipy/signal/tests/test_windows.py
@@ -502,6 +502,7 @@ class TestKaiser:
 class TestKaiserBesselDerived:
 
     def test_basic(self, xp):
+        # cover case `M < 1`
         w = windows.kaiser_bessel_derived(0.5, beta=4.0, xp=xp)
         xp_assert_equal(w, xp.asarray([]))
 

--- a/scipy/signal/windows/_windows.py
+++ b/scipy/signal/windows/_windows.py
@@ -1379,7 +1379,7 @@ def kaiser_bessel_derived(M, beta, *, sym=True, xp=None, device=None):
             "shapes"
         )
     elif M < 1:
-        return xp.array([])
+        return xp.asarray([])
     elif M % 2:
         raise ValueError(
             "Kaiser-Bessel Derived windows are only defined for even number "

